src/grt/src/GlobalRouter.cpp

```diff
@@ -2040,6 +2042,52 @@ void GlobalRouter::addLocalConnections(NetRouteMap& routes)
   }
 }
 
+bool GlobalRouter::pinOverlapsGSegment(const odb::Point& pin_position,
+                                       const int pin_layer,
+                                       const std::vector<odb::Rect>& pin_boxes,
+                                       const GRoute& route)
+{
+  bool segment_overlaps_pin = false;
+
+  // check if pin position on grid overlaps with the pin shape
```

Comment:
Are these two checks supposed to be both required or either required?  segment_overlaps_pin  may get overwritten in the 2nd check.